src/mpid/ch4/netmod/ofi/ofi_init.c

```diff
@@ -953,18 +953,79 @@ int MPIDI_OFI_get_vci_attr(int vci)
     return MPIDI_VCI_TX | MPIDI_VCI_RX;
 }
 
+/* Linked list internally used to keep track of
+ * allocated memory for which memory binding is
+ * requested by the user. */
+typedef struct mem_node {
+    void *ptr;
+    size_t size;
+    struct mem_node *next;
+} mem_node_t;
+
+static mem_node_t *mem_list_head = NULL;
+static mem_node_t *mem_list_tail = NULL;
+
 void *MPIDI_OFI_mpi_alloc_mem(size_t size, MPIR_Info * info_ptr)
 {
 
     void *ap;
-    ap = MPL_malloc(size, MPL_MEM_USER);
+    MPIR_hwtopo_type_e mem_type;
```

Comment:
Is there any difference between this and `MPIDIG_mpi_alloc_mem`? If not, this function should just call the generic version for now. So should ch4/ucx.

Reply:
No difference. ch4/ucx already calls the generic version. OFI had the code duplicated for some reason so in the draft I also duplicated it. I will just replace everything with `MPIDIG_mpi_alloc_mem`.